fairlearn/reductions/_exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -61,9 +61,9 @@ def __init__(self, estimator, constraints, eps=0.01, T=50, nu=None, eta_mul=2.0)
         self._nu = nu
         self._eta_mul = eta_mul
 
-        self._best_classifier = None
+        self._best_predictor = None
```

Comment:
I think that calling this `predictor` is confusing, because it is not a predictor in the `sklearn` sense. I don't think that we actually need to return this object. Let's get rid of `self._best_predictor` altogether and merge its functionality directly into `_pmf_predict` (we also don't need the `_mean_pred` wrapper).

Reply:
You're right, of course! One more of the remnants we've wanted to address in a while. I'll fix that...

Reply:
I just realized that `lagrangian.classifiers` only gets set when it's a new best value. So `len(lagrangian.classifiers)` can be less than `_last_t`. So I guess what I have right now with best_predictor being located at `_best_t` within `_predictors` (which is equivalent to `lagrangian.classifiers`) is not always correct. Is it simply the last entry in `lagrangian.classifiers`?

Reply:
... or it may be `expgrad._hs[expgrad._best_t]`. `_hs` has an entry for every iteration